components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/GatewayStartupListener.java

```diff
@@ -99,4 +102,40 @@ public void invoke() {
         }
     }
 
-}
+
+    public void startListener() {
+        new Thread(this).start();
+    }
+
+    @Override
+    public void run() {
+        deployArtifactsInGateway();
+    }
+
+    private void deployArtifactsInGateway() {
+
+        long retryDuration =
+                gatewayArtifactSynchronizerProperties.getRetryDuartion();
+        double reconnectionProgressionFactor = 2.0;
+        long maxReconnectDuration = 1000 * 60 * 60; // 1 hour
+
+        while (true) {
+            boolean isArtifactsDeployed = deployArtifactsAtStartup();
+            if (isArtifactsDeployed) {
+                log.info("Synapse Artifacts deployed Successfully in the Gateway");
+                break;
+            } else {
+                retryDuration = (long) (retryDuration * reconnectionProgressionFactor);
+                if (retryDuration > maxReconnectDuration) {
+                    retryDuration = maxReconnectDuration;
+                }
+                log.error("Unable to deploy synapse artifacts at gateway. Next retry in " + (retryDuration / 1000)
+                        + " seconds");
+                try {
+                    Thread.sleep(retryDuration);
+                } catch (InterruptedException ignore) {
+                }
+            }
+        }
+    }
+}
```

Comment:
new line in the end of file

Reply:
https://github.com/wso2/carbon-apimgt/pull/9546/commits/97796414dcecd5806df827b070c91470386312ea